renderdoccmd/renderdoccmd_linux.cpp

```diff
@@ -195,7 +236,319 @@ void VerifyVulkanLayer(const GlobalEnvironment &env, int argc, char *argv[])
+  keyboard_handle_enter,
+  keyboard_handle_leave,
+  keyboard_handle_key,
+  keyboard_handle_modifiers,
+};
+
+static void seat_handle_capabilities(void *data, struct wl_seat *seat,
+                                      uint32_t caps)
+{
+  if (((enum wl_seat_capability) caps & WL_SEAT_CAPABILITY_POINTER)
+      && !pointer)
+  {
+    pointer = wl_seat_get_pointer(seat);
+    wl_pointer_add_listener(pointer, &pointer_listener, NULL);
+  }
+  else if (!((enum wl_seat_capability) caps & WL_SEAT_CAPABILITY_POINTER)
+            && pointer)
+  {
+    wl_pointer_destroy(pointer);
+    pointer = NULL;
+  }
+
+  if (((enum wl_seat_capability) caps & WL_SEAT_CAPABILITY_KEYBOARD)
+      && !keyboard)
+  {
+    keyboard = wl_seat_get_keyboard(seat);
+    wl_keyboard_add_listener(keyboard, &keyboard_listener, NULL);
+  }
+  else if (!((enum wl_seat_capability) caps & WL_SEAT_CAPABILITY_KEYBOARD)
+            && keyboard)
+  {
+    wl_keyboard_destroy(keyboard);
+    keyboard = NULL;
+  }
+}
+
+static const struct wl_seat_listener seat_listener = {
+  seat_handle_capabilities,
+};
+
+static void registry_handler(void *data, struct wl_registry *registry,
+                              uint32_t id, const char *interface,
+                              uint32_t version)
+{
+  if (strcmp(interface, "wl_compositor") == 0)
+  {
+    compositor = (wl_compositor *) wl_registry_bind(registry, id,
+                  &wl_compositor_interface, 1);
+  }
+  else if (strcmp(interface, "wl_shell") == 0)
+  {
+    shell = (wl_shell *) wl_registry_bind(registry, id, &wl_shell_interface, 1);
+  }
+  else if (strcmp(interface, "wl_shm") == 0)
+  {
+    shm = (wl_shm *) wl_registry_bind(registry, id, &wl_shm_interface, 1);
+    wl_shm_add_listener(shm, &shm_listener, NULL);
+
+    cursor_theme = wl_cursor_theme_load(NULL, 32, shm);
+    default_cursor = wl_cursor_theme_get_cursor(cursor_theme, "left_ptr");
+  }
+  else if (strcmp(interface, "wl_seat") == 0)
+  {
+    seat = (wl_seat *) wl_registry_bind(registry, id, &wl_seat_interface, 1);
+    wl_seat_add_listener(seat, &seat_listener, NULL);
+  }
+}
+
+static void registry_remover(void *data, struct wl_registry *registry,
+                              uint32_t id)
+{
+}
+
+static const struct wl_registry_listener registry_listener = {
+  registry_handler,
+  registry_remover
+};
+
+static int set_cloexec_or_close(int fd)
+{
+  long flags;
+
+  if (fd == -1)
+    return -1;
+
+  flags = fcntl(fd, F_GETFD);
+  if (flags == -1)
+    goto err;
+
+  if (fcntl(fd, F_SETFD, flags | FD_CLOEXEC) == -1)
+    goto err;
+
+  return fd;
+
+err:
+  close(fd);
+  return -1;
+}
+
+static int create_tmpfile_cloexec(char *tmpname)
+{
+  int fd;
+
+#ifdef HAVE_MKOSTEMP
+  fd = mkostemp(tmpname, O_CLOEXEC);
+  if (fd >= 0)
+    unlink(tmpname);
+#else
+  fd = mkstemp(tmpname);
+  if (fd >= 0) {
+    fd = set_cloexec_or_close(fd);
+    unlink(tmpname);
+  }
+#endif
+
+  return fd;
+}
+
+static int os_create_anonymous_file(off_t size)
```

Comment:
We should not be creating any files on disk, even temporary, for this preview window. If memory is needed for a buffer then please allocate it with `new[]`.